js/src/Ice/ConnectionI.js

```diff
@@ -1140,9 +1154,17 @@ export class ConnectionI {
         return true;
     }
 
+    /**
+     * Sends the next message in the send queue, if any, and returns an array of messages that have already been sent and
+     * for which a reply has been received. Messages that have been sent are removed from the send queue.
+     *
+     * @returns An array of messages that have been sent and for which a reply has been received, or null if there are no
+     * completed messages.
+     */
     sendNextMessage() {
```

Comment:
This the return in the catch block of the try { while {} } loop correct now? Won't this just return undefined. Additionally, does generally catching LocalException here make sense?

Reply:
need to fix the return.

The catch is fine similar to the C# version

https://github.com/zeroc-ice/ice/blob/34ce55cb67eee969459b618bfe7e9a5cab91e2c9/csharp/src/Ice/ConnectionI.cs#L2095-L2098

Here is a bit more convoluted because in JavaScript there is not catch by type, maybe we should just add an assert to ensure we only catch LocalException, no other exception should appear here.